inc/zoo/swar/SWAR.h

```diff
@@ -231,6 +255,19 @@ template<int NBits, typename T>
 struct BooleanSWAR: SWAR<NBits, T> {
     using Base = SWAR<NBits, T>;
 
+    constexpr auto toMsbBools(const bool (&values)[Base::Lanes]) {
+        constexpr auto msbOfFirstLane = T{1} << (NBits - 1);
+        auto result = T{0};
+        for (auto arg : values) {
+            auto bit = arg ? msbOfFirstLane : 0;
+            result = (result << NBits) | bit;
+        }
+        return BooleanSWAR{result};
+    }
+
+    template <std::size_t N, typename = std::enable_if_t<N == Base::Lanes, T>>
+    constexpr BooleanSWAR(Literals_t<NBits, T>, const bool (&values)[N]) : Base(toMsbBools(values)) {}
```

Comment:
Please never follow this antipattern of doing all of the work in the derived class independently of the base class.
Always tie operations in the derived class to the operations in the base class, otherwise, the base class can drift out of alignment.
For example, you may
`Base(Literals<NBits, T>, values) { this->m_v << (NBits - 1); }`
When you have a hierarchical relation between pieces of code, do not fail to make the code in a way in which the relation is fully apparent;; this is the only way to keep the different pieces from eventually violating the relation given enough evolution of the code

Reply:
Yeah thanks for this feedback, I was also not happy with this for this reason. Will revise.

Reply:
@thecppzoo see what you think of this?